swift/example_code/s3/binary-streaming/Sources/streamdown/streamdown.swift

```diff
@@ -0,0 +1,165 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+/// A simple example that shows how to use the AWS SDK for Swift to
+/// upload files using binary streaming.
+
+// snippet-start:[swift.s3.streaming-down.imports]
+import ArgumentParser
+import AWSClientRuntime
+import AWSS3
+import Foundation
+import Smithy
+import SmithyHTTPAPI
+import SmithyStreams
+
+// snippet-end:[swift.s3.streaming-down.imports]
+
+// -MARK: - Async command line tool
+
+struct ExampleCommand: ParsableCommand {
+    // -MARK: Command arguments
+    @Option(help: "Name of the Amazon S3 bucket to download from")
+    var bucket: String
+    @Option(help: "Key of the file on Amazon S3 to download")
+    var key: String
+    @Option(help: "Local path to download the file to")
+    var dest: String?
+    @Option(help: "Name of the Amazon S3 Region to use (default: us-east-1)")
+    var region = "us-east-1"
+
+    static var configuration = CommandConfiguration(
+        commandName: "streamdown",
+        abstract: """
+        This example shows how to use binary data streaming to download a file
+        from Amazon S3.
+        """,
+        discussion: """
+        """
+    )
+
+    // snippet-start:[swift.s3.streaming-down]
+    /// Download a file from the specified bucket.
+    ///
+    /// - Parameters:
+    ///   - bucket: The Amazon S3 bucket name to get the file from.
+    ///   - key: The name (or path) of the file to download from the bucket.
+    ///   - destPath: The pathname on the local filesystem at which to store
+    ///     the downloaded file.
+    func downloadFile(bucket: String, key: String, destPath: String?) async throws {
+        let fileURL: URL
+
+        // If no destination path was provided, use the key as the name to use
+        // for the file in the downloads folder.
+        
+        if destPath == nil {
+            do {
+                try fileURL = FileManager.default.url(
+                    for: .downloadsDirectory,
+                    in: .userDomainMask,
```

Comment:
This is a me learning Swift thing - where does the `.` refer? Where are `downloadsDirectory` and `userDomainMask`?

Reply:
These are inferred by the Swift compiler based on the expected type. Since the `for:` parameter expects a `FileManager.SearchPathDirectory` value, that type is inferred. You could thus also write this as `FileManager.SearchPathDirectory.downloadsDirectory`, but that's cumbersome so you usually don't. Same kind of thing for the other parameters.